src/services/vault/client.rs

```diff
@@ -0,0 +1,141 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Auth {
+    pub client_token: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Login {
+    pub auth: Auth,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FetchSecretsData {
+    pub data: HashMap<String, String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FetchSecrets {
+    pub data: FetchSecretsData,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct VerifyTokenData {
+    pub expire_time: String,
+    pub issue_time: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct VerifyToken {
+    pub data: VerifyTokenData,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct UpdateSecret {
+    wrap_info: Option<String>,
+    warnings: Option<String>,
+    auth: Option<Auth>,
+}
+
+mod api_vault_url {
+    pub const LOGIN: &str = "https://bunker.mindvalley.dev:8200/v1/auth/okta/login";
+    pub const VERIFY_TOKEN: &str = "https://bunker.mindvalley.dev:8200/v1/auth/token/lookup-self";
+    pub const FETCH_SECRETS: &str = "https://bunker.mindvalley.dev:8200/v1/secret/data";
+    pub const PATCH_SECRET: &str = "https://bunker.mindvalley.dev:8200/v1/secret/data";
+}
+
+pub struct VaultClient {}
+
+impl VaultClient {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub async fn login(&self, email: &str, password: &str) -> Result<Login, reqwest::Error> {
+        let url = format!(
+            "{base_url}/{email}",
+            base_url = api_vault_url::LOGIN,
+            email = email
+        );
+
+        let client = reqwest::Client::new();
```

Comment:
Should instead build the client inside `new` and then reuse it with `self.client`?

Reply:
Thats a good catch, yeah I shoud've done that